ndcube/ndcube.py

```diff
@@ -948,6 +948,28 @@ def __rmul__(self, value):
     def __truediv__(self, value):
         return self.__mul__(1/value)
 
+    def __pow__(self, value):
+        new_data = self.data ** value
+        new_unit = self.unit if self.unit is None else self.unit ** value
+        new_uncertainty = self.uncertainty
+
+        if self.uncertainty is not None:
+            try:
+                new_uncertainty = new_uncertainty.propagate(np.power, self, self.data ** value, correlation=1)
+            except ValueError as e:
+                if "unsupported operation" in e.args[0]:
```

Comment:
This is gross, we should open an issue on astropy to raise different error classes for different errors.